### PR TITLE
#104

### PR DESCRIPTION
**Pull request by sava17**

### CODE REVIEW

File: src/main/java/ezuino/Main.java [hunk 1]
```diff
@@ -83,11 +83,11 @@ public static void main(String[] args) throws IOException {
...
         
-        FuncStructureVisitor fsv = new FuncStructureVisitor();
-        astNode.accept(fsv);
+        //FuncStructureVisitor fsv = new FuncStructureVisitor();
+        //astNode.accept(fsv);
  
-        ListVisitor lv = new ListVisitor();
-        astNode.accept(lv);
+        //ListVisitor lv = new ListVisitor();
+        //astNode.accept(lv);
```
Comment (joandrsn) [inline src/main/java/ezuino/Main.java]: Keep these lines

File: src/main/code.ezuino [hunk 2]
```diff
@@ -1,6 +1,3 @@
-func int main() {
-    while(1<2){
-        int a
-    }
-    return 1
-}
+int b
+int a
+b := a()
```
Comment (joandrsn) [inline src/main/code.ezuino]: Dont include in pull request

File: src/main/java/astvisitors/SymbolTableVisitor.java [hunk 3]
```diff
@@ -12,27 +12,33 @@
...
-        this.symbolTableHandler = new SymbolTableHandler(false);
+        this.symbolTableHandlerVariables = new SymbolTableHandler(false);
+        this.symbolTableHandlerFunctions = new SymbolTableHandler(false);
     }
 
     @Override
     public void visit(StartNode node) {
-        symbolTableHandler.openScope();
+        symbolTableHandlerVariables.openScope();
+        symbolTableHandlerFunctions.openScope();
```
Comment (joandrsn) [inline src/main/java/astvisitors/SymbolTableVisitor.java]: Could you make a `private void openScope` function which opens the scope of both symboltables?

File: src/main/java/astvisitors/SymbolTableVisitor.java [hunk 4]
```diff
@@ -12,27 +12,33 @@
 import symboltable.SymbolTableHandler;
 
 public class SymbolTableVisitor extends AstVisitor {
-    private SymbolTableHandler symbolTableHandler;
+    private SymbolTableHandler symbolTableHandlerVariables;
```
Comment (sentoa) [inline src/main/java/astvisitors/SymbolTableVisitor.java]: Suggestion
Refactor to shorter var names ~ stVariables?

File: src/main/java/astvisitors/SymbolTableVisitor.java [hunk 5]
```diff
@@ -12,27 +12,33 @@
...
     @Override
     public void visit(StartNode node) {
-        symbolTableHandler.openScope();
+        symbolTableHandlerVariables.openScope();
+        symbolTableHandlerFunctions.openScope();
         node.getDcls().accept(this);
         node.getStmts().accept(this);
-        symbolTableHandler.closeScope();
+        symbolTableHandlerVariables.closeScope();
+        symbolTableHandlerFunctions.closeScope();
```
Comment (joandrsn) [inline src/main/java/astvisitors/SymbolTableVisitor.java]: Could you make a `private void closeScope` function which closes the scope of both symboltables?

File: src/main/java/ezuino/Main.java [hunk 6]
```diff
@@ -83,11 +83,11 @@ public static void main(String[] args) throws IOException {
         astNode.acceptLevel(ipv, 0);
         //System.out.println(SymbolTableVisitor.symbolTableManager.getSymbolTableSize());
         
-        FuncStructureVisitor fsv = new FuncStructureVisitor();
-        astNode.accept(fsv);
+        //FuncStructureVisitor fsv = new FuncStructureVisitor();
```
Comment (sentoa) [inline src/main/java/ezuino/Main.java]: Should they be outcommented

File: src/main/java/astvisitors/SymbolTableVisitor.java [hunk 7]
```diff
@@ -42,23 +48,25 @@ public void visit(BlockNode node) {
...
         }
-        symbolTableHandler.closeScope();
+        symbolTableHandlerVariables.closeScope();
+        symbolTableHandlerFunctions.closeScope();
     }
 
     @Override
     public void visit(DclNode node) {
-        symbolTableHandler.enterSymbol(node.getID(), node);
+        System.out.println("Declaring variable" + node.getID());
```
Comment (sentoa) [inline src/main/java/astvisitors/SymbolTableVisitor.java]: Should there be a sysout?

**Pull request merged**